app/models/base_facility.rb

```diff
@@ -151,147 +150,7 @@ def find_facility_by_id(id)
-          facilities = facilities.where(unique_id: ids_for_type)
-        end
-        facilities.order('distance')
-      end.flatten
-    end
-    # rubocop:enable Metrics/ParameterLists
-
-    def zip_query(params)
-      # TODO: allow user to set distance from zip
-      zip_plus0 = params[:zip][0...5]
-      requested_zip = ZCTA.select { |area| area[0] == zip_plus0 }
-      # TODO: iterate over zcta, pushing each zip code that is within distance into an array
-      # TODO: change zip criteria to array of zip codes
-      conditions = "address ->'physical'->>'zip' ilike '#{requested_zip[0][0]}%'"
-      TYPES.flat_map do |facility_type|
-        get_facility_data(conditions, params[:type], facility_type, params[:services])
-      end
-    end
-
-    def build_result_set(bbox_num, type, services)
-      lats = bbox_num.values_at(1, 3)
-      longs = bbox_num.values_at(2, 0)
-      conditions = { lat: (lats.min..lats.max), long: (longs.min..longs.max) }
-      TYPES.flat_map { |facility_type| get_facility_data(conditions, type, facility_type, services) }
-    end
-
-    def ids_for_types(ids)
-      ids.split(',').each_with_object({}) do |type_id, obj|
-        facility_type, unique_id = type_id.split('_')
-        if facility_type && unique_id
-          obj[facility_type] ||= []
-          obj[facility_type].push unique_id
-        end
-      end
-    end
-
-    def build_result_set_from_ids(ids)
-      ids_for_types(ids).map do |facility_type, unique_ids|
-        klass = "Facilities::#{facility_type.upcase}Facility".constantize
-        klass.where(unique_id: unique_ids)
-      end
-    end
-
-    def get_facility_data(conditions, type, facility_type, services, additional_data = nil)
-      klass = TYPE_MAP[facility_type].constantize
-      return klass.none unless type.blank? || type == facility_type
-      klass = klass.select(additional_data) if additional_data
-      facilities = klass.where(conditions)
-      service_conditions = services&.map do |service|
-        service_condition(type, service)
-      end
-      facilities = facilities.where(service_conditions.join(' OR ')) if service_conditions&.any?
-      facilities = facilities.where.not(facility_type: DOD_HEALTH)
-      facilities
-    end
-
-    def service_condition(type, service)
-      case type
-      when 'benefits'
-        "services->'benefits'->'standard' @> '[\"#{service}\"]'"
-      when 'health'
-        "services->'health' @> '[{\"sl1\":[\"#{service}\"]}]'"
-      end
-    end
-
-    # Naive distance calculation, but accurate enough for map display sorting.
-    # If greater precision is ever needed, use Haversine formula.
-    def dist_from_center(bbox)
-      lambda do |facility|
-        center_x = (bbox[0] + bbox[2]) / 2.0
-        center_y = (bbox[1] + bbox[3]) / 2.0
-        Math.sqrt((facility.long - center_x)**2 + (facility.lat - center_y)**2)
-      end
+      FacilitiesQuery::Base.new(params).query
```

Comment:
i really like this new structure.  Did you think about making an module-level method for instantiating a FacilitiesQuery?  I feel like that can clean up some classes a bit.
So this line could be `FacilitiesQuery.generate_query(params).query` (or a better name) and that method would return the FacilitiesQuery object that ought to be used.  Then you can drop the `location_query_klass` and change `#run` to `#query` so it can duck-type.

Reply:
This line actually returns a relation, and I am now realizing the method name here should be `run` instead of `query` to match up what the child classes are doing

Reply:
you could call them all `get_relation`